src/Static.jl

```diff
@@ -481,8 +481,20 @@ Base.:(^)(x::Integer, y::True) = x
 Base.:(^)(x::BigInt, y::False) = one(x)
 Base.:(^)(x::BigInt, y::True) = x
 
+@inline function Base.ntuple(f::F, ::StaticInt{N}) where {F,N}
+    (N >= 0) || throw(ArgumentError(string("tuple length should be ≥ 0, got ", N)))
+    if @generated
+        quote
+            Base.Cartesian.@nexprs $N i -> t_i = f(i)
+            Base.Cartesian.@ncall $N tuple t
+        end
+    else
+        Tuple(f(i) for i = 1:N)
```

Comment:
```suggestion
            ntuple(f, Val(N))
```
any reason not to forward to this?

Reply:
I just copied the definition of `ntuple` from base so I'm assuming the compiler would just forward to this definition anyway if it chose the non-generated version